mlflow/store/dynamodb_store.py

```diff
@@ -0,0 +1,707 @@
+
+_DYNAMODB_ENDPOINT_URL_VAR = "MLFLOW_DYNAMODB_ENDPOINT_URL"
+_DYNAMODB_TABLE_PREFIX_VAR = "MLFLOW_DYNAMODB_TABLE_PREFIX"
+
+
+def _default_dynamodb_resource(endpoint_url=None):
+    dynamodb_endpoint_url = endpoint_url or get_env(_DYNAMODB_ENDPOINT_URL_VAR)
+    return boto3.resource('dynamodb', endpoint_url=dynamodb_endpoint_url)
+
+
+def _default_table_prefix():
+    return get_env(_DYNAMODB_TABLE_PREFIX_VAR) or "mlflow"
+
+
+def _dict_to_experiment(d):
+    return Experiment(
+        experiment_id=int(d['experiment_id']),
+        name=d['name'],
+        artifact_location=d.get('artifact_location') or '',
+        lifecycle_stage=d.get('lifecycle_stage', 'active')
+    )
+
+
+def _dict_to_run_info(d):
+    return RunInfo(
+        run_uuid=d['run_uuid'],
+        experiment_id=int(d['experiment_id']),
+        name=d.get('name') or '',
+        source_type=int(d.get('source_type') or 0),
+        source_name=d.get('source_name') or '',
+        entry_point_name=d.get('entry_point_name') or '',
+        user_id=d.get('user_id') or '',
+        status=int(d.get('status') or 0),
+        start_time=int(d.get('start_time') or 0),
+        end_time=int(d.get('end_time') or 0),
+        source_version=d.get('source_version') or '',
+        lifecycle_stage=d.get('lifecycle_stage', 'active'),
+        artifact_uri=d.get('artifact_uri') or '',
+    )
+
+
+def _list_to_run_tag(l):
+    return [RunTag(
+        key=d['key'],
+        value=d['value']
+    ) for d in l]
+
+
+def _list_to_run_param(l):
+    return [Param(
+        key=d['key'],
+        value=d['value']
+    ) for d in l]
+
+
+def _sort_metrics(l):
+    return sorted(l, key=lambda x: int(x['timestamp']))
+
+
+# Return a list of metrics with the first value (don't sort)
+def _list_to_run_metric(l):
+    return [Metric(key=rm['key'],
+            value=float(rm['metrics'][0]['value']),
+            timestamp=int(rm['metrics'][0]['timestamp'])) for rm in l]
+
+
+def _dict_to_run_metric_history(rm):
+    return [Metric(key=rm['key'], value=float(m['value']), timestamp=int(m['timestamp']))
+            for m in _sort_metrics(rm['metrics'])]
+
+
+def _entity_to_dict(obj):
+    return {k: None if v == '' else v for k, v in dict(obj).items()}
+
+
+def _filter_view_type(d, view_type=None):
+    return not view_type or view_type == ViewType.ALL or \
+        (view_type == ViewType.ACTIVE_ONLY and
+            d.get('lifecycle_stage') == RunInfo.ACTIVE_LIFECYCLE) or \
+        (view_type == ViewType.DELETED_ONLY and
+            d.get('lifecycle_stage') == RunInfo.DELETED_LIFECYCLE)
+
+
+def _filter_experiment(experiments, view_type=None, name=None):
+    return [e for e in experiments if _filter_view_type(e, view_type)
+            and (not name or e['name'] == name)]
+
+
+def _filter_run(runs, view_type=None, experiment_id=None):
+    return [r for r in runs if _filter_view_type(r, view_type)
+            and (not experiment_id or r['experiment_id'] == experiment_id)]
```

Comment:
can you use `experiment_id is None` instead of `not experiment_id`? Experiment id can be 0.